aws/components/ddb/setup.ftl

```diff
@@ -0,0 +1,886 @@
+                        ]
+                    ]
+                [#else]
+                    [@warn
+                        message="Ignoring linked backup regime \"${linkTargetCore.SubComponent.Name}\" that does not support tag based inclusion"
+                        context=linkTargetCore
+                    /]
+                [/#if]
+                [#break]
+        [/#switch]
+    [/#list]
+
+    [#local deletionPolicy = solution.Backup.DeletionPolicy]
+    [#local updateReplacePolicy = solution.Backup.UpdateReplacePolicy]
+
+    [#local ddsPreDeploySnapshotId = formatName(
+                                        ddsFullName,
+                                        (getCLORunId())?split('')?reverse?join(''),
+                                        "pre-deploy")]
+
+    [#local ddsTags = getOccurrenceCoreTags(occurrence, ddsFullName)]
+
+    [#local restoreSnapshotName = "" ]
+
+    [#if hibernate && hibernateStartUpMode == "restore" ]
+        [#local restoreSnapshotName = ddsPreDeploySnapshotId ]
+    [/#if]
+
+    [#local preDeploySnapshot = solution.Backup.SnapshotOnDeploy ||
+                            ( hibernate && hibernateStartUpMode == "restore" ) ||
+                            ddsManualSnapshot?has_content ]
+
+    [#if solution.AlwaysCreateFromSnapshot ]
+        [#if !ddsManualSnapshot?has_content ]
+            [@fatal
+                message="Snapshot must be provided to create this database"
+                context=occurrence
+                detail="Please provie a manual snapshot or a link to an RDS data set"
+            /]
+        [/#if]
+
+        [#local restoreSnapshotName = ddsManualSnapshot ]
+        [#local preDeploySnapshot = false ]
+
+    [/#if]
+
+    [#local dbParameters = {} ]
+    [#list solution.DBParameters as key,value ]
+        [#if key != "Name" && key != "Id" ]
+            [#local dbParameters += { key : value }]
+        [/#if]
+    [/#list]
+
+    [#local processorProfile    = getProcessor(occurrence, core.Type )]
+    [#local networkProfile      = getNetworkProfile(occurrence)]
+    [#local securityProfile     = getSecurityProfile(occurrence, core.Type )]
+    [#local requiredRDSCA       = securityProfile["SSLCertificateAuthority"]!"HamletFatal: SSLCertificateAuthority not found in security profile: " + solution.Profiles.Security ]
+
+    [#list solution.Links?values as link]
+        [#if link?is_hash]
+            [#local linkTarget = getLinkTarget(occurrence, link) ]
+
+            [@debug message="Link Target" context=linkTarget enabled=false /]
+
+            [#if !linkTarget?has_content]
+                [#continue]
+            [/#if]
+
+            [#local linkTargetCore = linkTarget.Core ]
+            [#local linkTargetConfiguration = linkTarget.Configuration ]
+            [#local linkTargetResources = linkTarget.State.Resources ]
+            [#local linkTargetAttributes = linkTarget.State.Attributes ]
+
+            [#if deploymentSubsetRequired("ddb", true)]
```

Comment:
Better to use DDS_COMPONENT_TYPE instead of ddb. Then if the type changes we don't have to track it down.

Reply:
Done